ResearchKit/Common/ORKStepViewController.m

```diff
@@ -66,9 +66,14 @@ @implementation ORKStepViewController {
 }
 
 - (void)initializeInternalButtonItems {
-    _internalContinueButtonItem = [[UIBarButtonItem alloc] initWithTitle:ORKLocalizedString(@"BUTTON_NEXT", nil) style:UIBarButtonItemStylePlain target:self action:@selector(goForward)];
-    _internalDoneButtonItem = [[UIBarButtonItem alloc] initWithTitle:ORKLocalizedString(@"BUTTON_DONE", nil) style:UIBarButtonItemStyleDone target:self action:@selector(goForward)];
-    _internalSkipButtonItem = [[UIBarButtonItem alloc] initWithTitle:ORKLocalizedString(@"BUTTON_SKIP", nil) style:UIBarButtonItemStylePlain target:self action:@selector(skip:)];
+    NSString *nextButtonTitle = [self localizedStringForString:ORKLocalizedString(@"BUTTON_NEXT", nil)];
+    NSString *doneButtonTitle = [self localizedStringForString:ORKLocalizedString(@"BUTTON_DONE", nil)];
+    NSString *skipButtonTitle = [self localizedStringForString:ORKLocalizedString(@"BUTTON_SKIP", nil)];
```

Comment:
@ghGarnik What is the intention here? `ORKLocalizedString` should already find the correct translation based on the user's preferred language (if included in ResearchKit.strings, which is likely).

If you then try to find a translation in `translations` based on that already localised string, it's likely to fail because:
a) we probably won't be providing translations for "Next", "Done" & "Skip" in the app.json.
b) the base language in app.json quite likely doesn't match the `ORKLocalizedString`.

I think our priority here should be to ensure that our translation logic resolves to the same language as `ORKLocalizedString`, so if we support French then both RK buttons and our strings appear in French, otherwise the RK buttons appear in French and our strings appear in the base language.

What do you think?

Reply:
@FWJonathan I had assumed that the only supported language would be `en`.
I've just modified localization logic, and turned it into `localizedStringForKey` . Now localization is taking into account both the current supported language and the base language (set on `CFBundleDevelopmentRegion`).
8c966ea54ef10d5cdbb58725f920250b846c3d74